pkg/scheduler/algorithm/priorities/selector_spreading.go

```diff
@@ -211,7 +211,16 @@ func filteredPod(namespace string, selector labels.Selector, nodeInfo *scheduler
 		return []*v1.Pod{}
 	}
 	for _, pod := range nodeInfo.Pods() {
-		if namespace == pod.Namespace && selector.Matches(labels.Set(pod.Labels)) {
+		if namespace != pod.Namespace {
+			continue
+		}
+		// Ignore pods being deleted for spreading purposes
+		// Similar to how it is done for SelectorSpreadPriority
+		if pod.DeletionTimestamp != nil {
```

Comment:
You could add this condition above the existing `if namespace...` and not change anything else. That would be simpler.

Reply:
I thought it would be better to test deletion timestamp only for pods in the same namespace instead for all pods to save a few cycles?